### PR TITLE
Fix MSVC build, cleanup solution folders

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: DoxygenRule.cmake [hunk 1]
```diff
@@ -109,28 +109,26 @@ add_custom_target(doxygen_html_${PROJECT_NAME}
...
 
-if(NOT TARGET doxygen)
-  add_custom_target(doxygen)
-  set_target_properties(doxygen PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+if(NOT TARGET doxygens)
+  add_custom_target(doxygens)
 endif()
-add_dependencies(doxygen ${PROJECT_NAME}-doxygen)
+add_dependencies(doxygens ${PROJECT_NAME}-doxygen)
```
Comment (eile) [inline DoxygenRule.cmake]: Needs some doc update?

**Comment by eile** [comment]
+1, although I'm not sure if 'targets' is a better name then 'target'

**Comment by tribal-tec** [comment]
To me, I'd rather remove those completely. A meta-target for 'branch' does more harm I suppose...

File: DoxygenRule.cmake (hunk 1 above)
Comment (tribal-tec) [inline DoxygenRule.cmake]: Does it? I'll check. I can also revert them. I don't like or use them anyways :p

Comment (rdumusc) [inline DoxygenRule.cmake]: Oh, since you're changing this, what about naming the targets 'documentation' instead of 'doxygen'?

**Comment by eile** [comment]
branch/tag for sure. I just used make doxygen to update all project's documentation.

File: GitExternal.cmake [hunk 2]
```diff
@@ -281,7 +281,7 @@ endif()")
 
           foreach(_target flatten_git_external_${GIT_EXTERNAL_NAME} flatten_git_external update_git_external_${GIT_EXTERNAL_NAME} ${GIT_EXTERNAL_TARGET} update_git_external update)
             set_target_properties(${_target} PROPERTIES
-              EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "git")
+              EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/git)
```
Comment (rdumusc) [inline GitExternal.cmake]: no quotes here?

Comment (tribal-tec) [inline GitExternal.cmake]: Why? PROJECT_NAME should not have spaces.

Comment (rdumusc) [inline GitExternal.cmake]: Right, it was just for consistency - if you scroll down all the next ones are quoted..

Comment (tribal-tec) [inline GitExternal.cmake]: Oh snap. I'll check them

**Comment by tribal-tec** [comment]
Updated

**Comment by eile** [comment]
+1

File: CpplintTargets.cmake [hunk 3]
```diff
@@ -26,13 +26,11 @@ endif(NOT CPPLINT_FOUND)
...
   add_custom_target(cpplint_${PROJECT_NAME} COMMENT "${CPPLINT_NOT_FOUND_MSG}")
   set_target_properties(cpplint_${PROJECT_NAME} PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "Tests/${PROJECT_NAME}/cpplint")
+    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/tests/cpplint)
 endif(NOT CPPLINT_FOUND)
 
 if(NOT TARGET cpplint)
   add_custom_target(cpplint)
-  set_target_properties(cpplint PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "Tests")
```
Comment (rdumusc) [inline CpplintTargets.cmake]: Are you sure? Should still be excluded from default build, no?

File: DoxygenRule.cmake [hunk 4]
```diff
@@ -108,26 +108,24 @@ add_custom_target(doxygen_html_${PROJECT_NAME}
...
-  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
 if(ENABLE_COVERAGE) # CoverageReport generated by "tests" in this case
   add_dependencies(${PROJECT_NAME}-doxygen coverage_${PROJECT_NAME})
 endif()
 
 if(NOT TARGET doxygen)
   add_custom_target(doxygen)
-  set_target_properties(doxygen PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
```
Comment (rdumusc) [inline DoxygenRule.cmake]: Same here?

File: clangcheckTargets.cmake [hunk 5]
```diff
@@ -18,8 +18,6 @@ endif()
 if(NOT CLANGCHECK)
   if(NOT TARGET clangcheck)
     add_custom_target(clangcheck COMMENT "clang-check executable not found")
-    set_target_properties(clangcheck PROPERTIES
-      EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "Tests")
```
Comment (rdumusc) [inline clangcheckTargets.cmake]: And here?

**Comment by rdumusc** [comment]
+1 otherwise! Next commit to remove "harmful" meta-targets like branch/tag/etc..?

File: CpplintTargets.cmake (hunk 3 above)
Comment (tribal-tec) [inline CpplintTargets.cmake]: True

**Pull request merged**